### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by KieranHons**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -132,3 +132,18 @@ $ npm run compile start
 xxx [m:15071] [info ] Opening WebSocket connection
 xxx [m:15071] [info ] WebSocket connection established. Listening for incoming messages
 ```
+
+### Testing
+
+In order for tests to run you'll need to have a running local hadron collider environment:
+https://github.com/absa-subatomic/local-hadron-collider
```
Comment (kbristow) [inline README.md]: So the unit tests should mock out any calls to lhc. What in particular is requiring the lhc to be running?

File: config/test.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "openshiftHost" : "",
+    "gluon": {
+      "baseUrl": "http://localhost:8080"
+    },
+    "docs": {
+      "baseUrl": "http://localhost:8080"
+    }
+  },
+  "teamId": "F7E6GGR63A",
+  "token": "88agdfgdf8afg7f7d88g9aeadga8sd89cj",
```
Comment (kbristow) [inline config/test.json]: I assume these are fake? :) @donovanmuller is there a better way than having a test config file committed to the repo that you would like?

File: package.json [hunk 3]
```diff
@@ -71,7 +73,7 @@
     "sendCodeCoverage": "nyc report --reporter=json && codecov -f coverage/*.json",
     "tscSourceMap": "tsc test/**/*.ts --module commonjs --sourcemap --lib es2015,dom,esnext.asynciterable",
     "start": "atomist start --no-install --no-compile",
-    "test": "nyc mocha --require espower-typescript/guess \"test/**/*.ts\"",
+    "test": "nyc mocha --require espower-typescript/guess \"test/**/*.ts\" --exclude \"test/gluon/team/DevOpsEnvironmentRequestedTest.ts\"",
```
Comment (kbristow) [inline package.json]: I presume this is the test requiring the lhc to be running?

File: src/gluon/team/DevOpsEnvironmentRequested.ts [hunk 4]
```diff
@@ -367,12 +367,11 @@ export class DevOpsEnvironmentRequested implements HandleEvent<any> {
...
                                 return SuccessPromise;
                             }, () => {
-                                return OCCommon.commonCommand("secrets new-sshauth",
+                                return OCCommon.commonCommand("create secret generic",
                                     "bitbucket-ssh",
                                     [],
                                     [
-                                        new SimpleOption("-ssh-privatekey", QMConfig.subatomic.bitbucket.cicdPrivateKeyPath),
-                                        new SimpleOption("-ca-cert", QMConfig.subatomic.bitbucket.caPath),
+                                        new SimpleOption("-from-file", QMConfig.subatomic.bitbucket.cicdPrivateKeyPath),
```
Comment (kbristow) [inline src/gluon/team/DevOpsEnvironmentRequested.ts]: Where are these changes coming from?

File: test/gluon/TestGraphClient.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public var: any;
+
+    public executeQueryFromFile<T, Q>(path: string, variables?: Q, options?: any, current?: string): Promise<any> {
+        this.path = path;
+        const json = {
+         ChatId: [
+             {
+             userId: "U967SDE6",
+             screenName: "Test.User", // Throws error with variables.userId
+            },
```
Comment (kbristow) [inline test/gluon/TestGraphClient.ts]: This seems very static. Perhaps there is a way to mock this out for your specific file queries. We can take a look when I get back.

File: test/gluon/jenkins/JenkinsBuildTest.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        subject.handle(fakeContext)
+            .then(() => {
+                logger.info(fakeContext.messageClient.textMsg);
+                //assert(fakeContext.messageClient.textMsg.text.trim() === `🚀 *${applicationName}* is being built...`);
+                return Promise.resolve();
+            })
+            .then(done, done);
+    });
+});*/
```
Comment (kbristow) [inline test/gluon/jenkins/JenkinsBuildTest.ts]: What is this file? :)

File: README.md (hunk 1 above)
Comment (kbristow) [reply README.md]: Probably the oc commands?

File: test/gluon/team/DevOpsEnvironmentTest.ts [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+        subject.handle(fakeContext)
+            .then(() => {
+                return Promise.resolve();
+            })
+            .then(() => {
+                return Promise.resolve();
+            })
+            .then(() => {
+                return Promise.resolve();
```
Comment (kbristow) [inline test/gluon/team/DevOpsEnvironmentTest.ts]: What do these multiple thens do?

File: README.md (hunk 1 above)
Comment (KieranHons) [reply README.md]: Yeah the oc commands were causing the tests to fail.

File: config/test.json (hunk 2 above)
Comment (KieranHons) [reply config/test.json]: Yeah totally fake :) tomorrow we can have a break away to discuss this

File: README.md (hunk 1 above)
Comment (kbristow) [reply README.md]: Cool, we can mock those out too. Lets take a look at some mocking libraries tomorrow.

File: src/gluon/team/DevOpsEnvironmentRequested.ts (hunk 4 above)
Comment (KieranHons) [reply src/gluon/team/DevOpsEnvironmentRequested.ts]: Wanted to chat about this, this current command is deprecated. So thought I'll upgrade

File: test/gluon/TestGraphClient.ts (hunk 5 above)
Comment (KieranHons) [reply test/gluon/TestGraphClient.ts]: Oki shweet

File: test/gluon/team/DevOpsEnvironmentTest.ts (hunk 7 above)
Comment (KieranHons) [reply test/gluon/team/DevOpsEnvironmentTest.ts]: Its part of the nesting, not ideal but only way I could get it to work

File: test/gluon/jenkins/JenkinsBuildTest.ts (hunk 6 above)
Comment (kbristow) [reply test/gluon/jenkins/JenkinsBuildTest.ts]: Did you take a look at this?

File: src/gluon/team/DevOpsEnvironmentRequested.ts (hunk 4 above)
Comment (kbristow) [reply src/gluon/team/DevOpsEnvironmentRequested.ts]: Did you test this? If so I will push it through.

**Pull request merged**